desc/magnetic_fields.py

```diff
@@ -1637,6 +1644,351 @@ def from_surface(
+            "Typical helicity (M,N) has N=NFP.",
+        )
+
+    def change_resolution(
+        self,
+        L_well=None,
+        M_well=None,
+        L_shift=None,
+        M_shift=None,
+        N_shift=None,
+        NFP=None,
+    ):
+        """Set the spectral resolution of well and shift terms.
+
+        Parameters
+        ----------
+        L_well : int
+            Radial resolution of the magnetic well parameters B_lm.
+        M_well : int
+            Number of spline points in the magnetic well parameters B_lm.
+        L_shift : int
+            Radial resolution of x_lmn.
+        M_shift : int
+            Poloidal resolution of x_lmn.
+        N_shift : int
+            Toroidal resolution of x_lmn.
+        NFP : int
+            Number of field periods.
+
+        """
+        self._NFP = setdefault(NFP, self.NFP)
+        self._L_well = setdefault(L_well, self.L_well)
+        self._M_well = setdefault(M_well, self.M_well)
+        self._L_shift = setdefault(L_shift, self.L_shift)
+        self._M_shift = setdefault(M_shift, self.M_shift)
+        self._N_shift = setdefault(N_shift, self.N_shift)
+
+        old_modes_well = self.well_basis.modes
+        old_modes_shift = self.shift_basis.modes
+
+        self.well_basis.change_resolution(self.L_well)
+        self.shift_basis.change_resolution(
+            self.L, self.M, self.N, NFP=self.NFP, sym="cos(t)"
+        )
+
+        self._x_lmn = copy_coeffs(self.x_lmn, old_modes_shift, self.shift_basis.modes)
+
+        old_well = self.B_lm.reshape((-1, self.M_well))
+        new_well = np.zeros((self.L_well + 1, self.M_well))
+        for j in range(self.M_well):
+            new_well[:, j] = copy_coeffs(
+                old_well[:, j], old_modes_well, self.well_basis.modes
+            )
+        self._B_lm = new_well.flatten()
+
+    # TODO: reuse compute function?
+    def compute_well(self, rho, eta):
+        """Compute well shape from B_lm.
+
+        Parameters
+        ----------
+        rho : ndarray
+            Radial coordinates.
+        eta : ndarray
+            Coordinates along the field line.
+
+        Returns
+        -------
+        B : ndarray
+            Magnetic field strength.
+
+        """
+        alpha = np.zeros_like(rho)
+        nodes = np.array([rho, eta, alpha]).T
+
+        # reshaped to size (L_well, M_well)
+        B_lm = self.B_lm.reshape((self.well_basis.L + 1, -1))
+        # assuming single flux surface, so only take first row (single node)
+        B_input = (self.well_basis.evaluate(nodes) @ B_lm)[0, :]
+        B_input = np.sort(B_input)  # sort to ensure monotonicity
+        eta_input = np.linspace(0, jnp.pi / 2, num=B_input.size)
+
+        # |B|_omnigeneous is an even function so B(-eta) = B(+eta) = B(|eta|)
+        B = interp1d(np.abs(eta), eta_input, B_input, method="monotonic-0")
+        return B
+
+    # TODO: move to compute function?
+    def compute_boozer_angles(self, rho, eta, alpha, iota):
```

Comment:
same as above

Reply:
could we make iota a kwarg? and in the outer compute method we pass in the correct value?

Reply:
removed and made part of the regular compute API